wallet/src/signer/ledger_signer/ledger_messages.rs

```diff
@@ -0,0 +1,367 @@
+        )
+        .await?;
+    }
+
+    let mut resp = vec![];
+    let num_outputs = outputs.len();
+    for (idx, o) in outputs.into_iter().enumerate() {
+        if idx < num_outputs - 1 {
+            send_chunked_expect_empty_ok_response(
+                ledger,
+                Ins::SIGN_TX,
+                P1SignTx::Output.into(),
+                &ledger_encode(SignTxReq::Output(o)),
+            )
+            .await?;
+        } else {
+            // the response from the last output will have the first signature returned
+            resp = send_chunked(
+                ledger,
+                Ins::SIGN_TX,
+                P1SignTx::Output.into(),
+                &ledger_encode(SignTxReq::Output(o)),
+            )
+            .await?;
+        }
+    }
+
+    let mut signatures: BTreeMap<_, Vec<_>> = BTreeMap::new();
+
+    let next_sig = ledger_encode(SignTxReq::NextSignature);
+    let mut msg_buf = vec![APDU_CLASS, Ins::SIGN_TX, P1SignTx::NextSignature.into(), P2_DONE];
+    msg_buf.push(next_sig.len() as u8);
+    msg_buf.extend(next_sig);
+    loop {
+        let SignatureResult {
+            sig,
+            input_idx,
+            has_more_signatures,
+        } = decode_signature_response(&resp)?;
+
+        signatures.entry(input_idx).or_default().push(sig);
+
+        if !has_more_signatures {
+            break;
+        }
+
+        resp = exchange_message(ledger, &msg_buf).await?;
+    }
+
+    Ok(signatures)
+}
+
+fn decode_signature_response(resp: &[u8]) -> Result<SignatureResult, LedgerError> {
+    let input_idx = *resp.first().ok_or(LedgerError::InvalidResponse)? as usize;
+    let has_more_signatures = *resp.last().ok_or(LedgerError::InvalidResponse)? == P2_MORE;
+
+    let sig: LedgerSignature =
+        ledger_decode_all(&resp[..resp.len() - 1][1..]).ok_or(LedgerError::InvalidResponse)?;
+
+    Ok(SignatureResult {
+        sig,
+        input_idx,
+        has_more_signatures,
+    })
+}
+
+pub fn to_ledger_tx_output(value: &chain::TxOutput) -> LTxOutput {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
+
+pub fn to_ledger_amount(value: &primitives::Amount) -> LAmount {
+    LAmount::from_atoms(value.into_atoms())
+}
+
+pub fn to_ledger_outpoint(value: &chain::UtxoOutPoint) -> LUtxoOutPoint {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
+
+pub fn to_ledger_account_outpoint(value: &chain::AccountOutPoint) -> LAccountOutPoint {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
+
+pub fn to_ledger_account_nonce(value: &chain::AccountNonce) -> LAccountNonce {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
+
+pub fn to_ledger_account_command(value: &chain::AccountCommand) -> LAccountCommand {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
+
+pub fn to_ledger_order_account_command(value: &chain::OrderAccountCommand) -> LOrderAccountCommand {
+    ledger_decode_all(value.encode().as_slice()).expect("ok")
+}
```

Comment:
This is not great. Perhaps we should move the code from `common/tests/primitives_repo_consistency/utils/converters.rs` from tests to the `common` crate itself and use it here.But note that we'll need to get rid of the panic for TokenV0 inside the converters and return an error instead. Which means that `ConvertFrom/convert_from` should become `TryConvertFrom/try_convert_from` and return a `Result`. The error type for the result should IMO be a separate type containing only one error variant.

Also, the "L" aliases for the primitives repo types are not needed IMO (or at least they should be "P", because they are specific not to the Ledger, but to the primitives repo).